src/AzureDeprecation.APIs.REST/Functions/GetDeprecationsFunction.cs

```diff
@@ -1,34 +1,38 @@
+using AutoMapper;
+using AzureDeprecation.APIs.REST.DataAccess.Contracts;
+using AzureDeprecation.APIs.REST.Utils;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.WebJobs;
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
+using Presentation = AzureDeprecation.APIs.REST.Contracts;
 
 namespace AzureDeprecation.APIs.REST.Functions
 {
-    public static class GetDeprecationsFunction
+    public class GetDeprecationsFunction
     {
+        readonly IDeprecationsRepository _deprecationsRepository;
+        readonly IMapper _mapper;
+
+        public GetDeprecationsFunction(
+            IDeprecationsRepository deprecationsRepository,
+            IMapper mapper)
+        {
+            _deprecationsRepository = deprecationsRepository;
+            _mapper = mapper;
+        }
+        
         [FunctionName("get-deprecations")]
-        public static async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "get", Route = "api/v1/deprecations")] HttpRequest request,
-            ILogger log)
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "api/v1/deprecations")] HttpRequest request,
+            ILogger log,
+            CancellationToken cancellationToken = default)
         {
-            log.LogInformation("C# HTTP trigger function processed a request.");
-
-            var name = request.Query.TryGetValue("name", out var n) ? n.FirstOrDefault() : null;
-
-            if (name is null)
-            {
-                string requestBody = await new StreamReader(request.Body).ReadToEndAsync();
-                dynamic? data = JsonConvert.DeserializeObject(requestBody);
-                name = data?.name;
-            }
-
-            string responseMessage = string.IsNullOrEmpty(name)
-                ? "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response."
-                : $"Hello, {name}. This HTTP triggered function executed successfully.";
-
-            return new OkObjectResult(responseMessage);
+            var filter = DeprecationRequestModelBinder.CreateModel(request.Query);
+            var entities = await _deprecationsRepository.GetDeprecationsAsync(filter, cancellationToken);
+            var result = _mapper.Map<Presentation.DeprecationNoticesResponse>(entities);
+            return new JsonResult(result, JsonSettingsProvider.GetDefault());
```

Comment:
Can't we just do this?
```suggestion
            return new OkResult(result);
```

Reply:
OkResult uses a default serializer which lead to serializing enums as integers. 
I wasn't able to find a better way to avoid this ~stupid~ behavior.

Reply:
can you do try managing it in the services? I think `options.AddMvcCore().Jsonsomething`

Reply:
https://github.com/azure-deprecation/infrastructure/blob/migrate-deprecation-persistance/src/AzureDeprecation.Runtimes.AzureFunctions/Startup.cs#L54-L55

Reply:
Looks like this stuff doesn't work for azure function responses. Had to add attributes manually